src/Network/GRPC/MQTT/Logging.hs

```diff
@@ -0,0 +1,36 @@
+{-
+  Copyright (c) 2021 Arista Networks, Inc.
+  Use of this source code is governed by the Apache License 2.0
+  that can be found in the COPYING file.
+-}
+module Network.GRPC.MQTT.Logging where
+
+import Relude
+
+data Logger = Logger
+  { log :: Text -> IO ()
+  , verbosity :: Verbosity
+  }
+
+data Verbosity
+  = Silent
+  | Error
+  | Warn
+  | Info
+  | Debug
+  deriving (Show, Eq, Enum, Ord)
+
+noLogging :: Logger
+noLogging = Logger (\_ -> pure ()) Silent
+
+logErr :: Logger -> Text -> IO ()
+logErr = logVerbosity Error
+logWarn :: Logger -> Text -> IO ()
+logWarn = logVerbosity Warn
+logInfo :: Logger -> Text -> IO ()
+logInfo = logVerbosity Info
+logDebug :: Logger -> Text -> IO ()
+logDebug = logVerbosity Debug
+
+logVerbosity :: Verbosity -> Logger -> Text -> IO ()
+logVerbosity v logger msg = when (verbosity logger >= v) $ log logger msg
```

Comment:
I think you may need to set LineBuffering for stderr and stdout, I don't think it's the default. Our internal awake haskell logging library uses https://hackage.haskell.org/package/fast-logger and I wonder if that might be appropriate to use here, too.

Reply:
Here we don't know if the logging is going to stderr/stdout. The specific logging function comes from the user. So, it would be up to the user to set the LineBuffering if the logs are being printed there.

Reply:
Oh I see, okay.